aws/resource_aws_ecr_repository.go

```diff
@@ -143,20 +147,24 @@ func resourceAwsEcrRepositoryDelete(d *schema.ResourceData, meta interface{}) er
 	}
 
 	log.Printf("[DEBUG] Waiting for ECR Repository %q to be deleted", d.Id())
+	input := &ecr.DescribeRepositoriesInput{
+		RepositoryNames: aws.StringSlice([]string{d.Id()}),
+	}
 	err = resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
-		_, err := conn.DescribeRepositories(&ecr.DescribeRepositoriesInput{
-			RepositoryNames: aws.StringSlice([]string{d.Id()}),
-		})
+		_, err = conn.DescribeRepositories(input)
 		if err != nil {
 			if isAWSErr(err, ecr.ErrCodeRepositoryNotFoundException, "") {
 				return nil
 			}
 			return resource.NonRetryableError(err)
 		}
 
-		return resource.RetryableError(
-			fmt.Errorf("%q: Timeout while waiting for the ECR Repository to be deleted", d.Id()))
+		return resource.RetryableError(fmt.Errorf("%q: Timeout while waiting for the ECR Repository to be deleted", d.Id()))
 	})
+	if isResourceTimeoutError(err) {
+		_, err = conn.DescribeRepositories(input)
+	}
```

Comment:
We need to ignore the not found exception here before returning it as an error since we are trying to determine if the repository was properly deleted here:

```suggestion
	}
	
	if isAWSErr(err, ecr.ErrCodeRepositoryNotFoundException, "") {
		return nil
	}
```